### ISSUE TITLE
Design Intent - Accessibility Feedback - Decision Reviews Team, Decision Review Forms, Downloadable PDF Copy of an Appeal

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#67083]
## VFS actions
- [x] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] Comment on this ticket:
  - [x] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [x] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [x] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [x] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- Really glad to see this work being done, and hopefully it's a model for other products moving forward. Not being able to review your submission after completing a form seems like a big missing piece!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Consider:** As I said in the touchpoint meeting, the benefits and challenges around PDFs all have to do with reputation. People tend to think of PDFs as being "official" and have a lot of trust in a downloadable PDF as a digital record of what they've done. But screen reader users encounter so many inaccessible PDFs that many of them just don't bother opening PDFs because their expectations have been so tainted. 
   
   No way to win here, except to:
   1. Make the PDF as accessible as possible
   2. If possible, provide information in multiple data formats --- eg. available as a PDF download, but also available as a webpage you can visit, or as a .txt file you can download, or something like that.
   
   That may not be possible in the current iteration of what you're doing, but it's something to have on your radar and explore when you can.
- **Must:** The unfilled PDFs that get generated by VA's form generator are reasonably accessible. The challenge for the filled form will be ensuring that the data you're adding to the PDF is added so that a screen reader user is able to review their submission and hear everything in a sensible order, eg. the "First Name" field label, then their first name, then the "Last Name" field label, and so on. I don't know enough about the technical side of populating the PDFs to know how difficult that's going to be, so my best advice here is to try to get some sample filled forms generated as soon as possible so you have lots of time to test and decide if some other approach needs to be explored.
- **Should:** The watermark is tricky to implement accessibly, because you need to simultaneously:
   - Make the contrast ratio between the watermark text and the white background at least 3:1 (large text against background)
   - Make the contrast ratio between the form text and the watermark at least 4.5:1 (regular text against background)
   
   It's a pretty narrow color window to aim for, especially if you're just in grayscale. And those ratios are minimums for WCAG conformance, so you still might end up with something that's kind of difficult to read for some users even if you meet those contrast ratios. My advice would be to explore something other than a watermark, like a stamp in the top margin of each page that communicates what the watermark is intending to communicate. Overlapping text is just hard to do accessibly.
   - **Conditional must:** If you need to do a watermark anyway...
      - Make sure there's some kind of screen reader friendly text matching the watermark that's announced at the beginning of the document, so that it's the first thing screen reader users hear when opening the document.
      - If you're not able to find an accessible color combination, try to position the watermark to minimize how much text it obscures, and if it has to obscure text make sure it's form labels or instructions and not user submissions. At minimum we want to make sure users can review what they've entered, even if they can't read all of the instruction text again.
- **Should:** In the email notification, I think it would be good to include some kind of copy indicating that users may need to re-authenticate to download the PDF.

**Comment by davidakennedy** [comment]
I did some brief testing on the filled PDF from Lighthouse. It has some challenges, which we would need to work through.

## Testing results

The challenges:

- The current PDF isn't tagged, which gives assistive technology an idea of what the structure of the document.
- It doesn't have a title for the document.
- It doesn't have a primary language set.
- The tab order isn't set.
- Some images lack alt text.
- Lots of issues with the structure of the tables, lists and headings in the PDF.

Some of these are easy fixes. Tagging it can be more involved, depending on how the PDF gets created.

## Next steps

- We should come in with a clean, tagged PDF, then figure out how the watermark may need to work.
- Connect with the Lighthouse team to figure out how the PDF gets created. PDFs get made on the fly elsewhere on VA.gov, and we may be able to leverage the same tooling for more accessible results.
- Investigate larger question: Is the watermark needed? The 1010CG has a downloadable PDF of form data with no watermark. Is there research present that suggest that Veterans aren't confused by it and/or the need to submit it?

**Comment by davidakennedy** [comment]
After our meeting Friday, exploring how PDFs are made via Lighthouse, I wanted to share some key aspects of accessible PDF creation and some potential software solutions.

## Key Aspects of PDF Accessibility

1. Searchable text: The document is real text.
2. Document Language and Title Indication: It has a language and title so assistive technology users can read what the document is about.
3. Alternative Text Descriptions for Non-Text Elements: Any logos, images or decorative images need to be marked as much with proper alternative text so assistive technology users know what the image means.
4. Tagged Structure: PDF documents use tags, similar to HTML, to let assistive technology understand what the purpose of different types of text (heading, list item, etc.).

These are the basics of PDF accessibility, and where most PDF creation software gets it wrong. More details are covered in [Adobe's guidance](https://www.adobe.com/accessibility/pdf/pdf-accessibility-overview.html) and [WebAIM's guidance](https://webaim.org/techniques/acrobat/).

## Possible Software

- [PDFReactor](https://www.pdfreactor.com/)
- [Prince](https://www.princexml.com/)
- [PDFKit](https://pdfkit.org/)

I don't think any of the above are perfect for our use case, but I still wanted to share them as what's good about them may help guide our choice in PDF library. A few more detailed thoughts on each of these:

- PDFKit uses JavaScript, Node and the browser. VA.gov [uses it already for some frontend PDF generation](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/pdf/README.md).
- Prince and PDFReactor are paid libraries.
- I believe all of these can generate PDFs from HTML as well as edit an existing PDF.
- Prince has a number of wrappers available for different server side languages.
